### PR TITLE
56993 - pos include cancelled products in large exchangeset

### PR DESCRIPTION
**Pull request by Vishal14583**
- Allowing POS to create exchange set with warning when cancelled products found in ESS response.
- Restrict POS to create exchange set if any other reason found except "noDataAvailableForCancelledProduct" reason.
- Publish Event ids as an artifact in pipeline.

### CODE REVIEW

**Review by mattwin1 — APPROVED** [review]
Reviewed Devops changes: addition of eventid scripts and yaml tasks + tf export of eventhub vars
LGTM, approved.

File: azure-pipelines.yml [hunk 1]
```diff
@@ -534,7 +547,7 @@ stages:
         value: $(AutoTestClientSecret_Authed)
       - name: "ESSApiConfiguration.EssClientId"
         value: $(ESSClientId)
-    condition: and(succeeded(), or(eq(variables['Build.SourceBranch'], 'refs/heads/main'),startsWith(variables['Build.SourceBranch'], 'refs/heads/release/')))
+#    condition: and(succeeded(), or(eq(variables['Build.SourceBranch'], 'refs/heads/main'),startsWith(variables['Build.SourceBranch'], 'refs/heads/release/')))
```
Comment (ajasonca) [inline azure-pipelines.yml:550]: Can I just confirm this condition is supposed to be removed for the QA deployment stage?

**Review by ajasonca — CHANGES_REQUESTED** [review]
A couple of minor changes, and want to confirm the removal of the QA Deploy stage condition.

File: UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Fulfilment/Services/FulfilmentDataService.cs [hunk 2]
```diff
@@ -174,15 +175,22 @@ private async Task<string> PostProductIdentifiersToESS(List<string> productIdent
         {
             ExchangeSetResponseModel exchangeSetResponseModel = await _essService.PostProductIdentifiersData(productIdentifiers);
 
-            if (exchangeSetResponseModel.RequestedProductsNotInExchangeSet.Count() == 0)
+            if (exchangeSetResponseModel.RequestedProductsNotInExchangeSet.Count() > 0)
```
[the remark this answers is not in the document]
Comment (sumod13943) [reply UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Fulfilment/Services/FulfilmentDataService.cs]: Changed.

File: UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Fulfilment/Services/FulfilmentDataService.cs [hunk 3]
```diff
@@ -230,18 +238,25 @@ private async Task<string> GetProductDataSinceDateTimeFromEss(string sinceDateTi
         {
             ExchangeSetResponseModel exchangeSetResponseModel = await _essService.GetProductDataSinceDateTime(sinceDateTime);
 
-            if (exchangeSetResponseModel.RequestedProductsNotInExchangeSet.Count() == 0)
+            if (exchangeSetResponseModel.RequestedProductsNotInExchangeSet.Count() > 0)
```
[the remark this answers is not in the document]
Comment (sumod13943) [reply UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Fulfilment/Services/FulfilmentDataService.cs]: Changed.

File: azure-pipelines.yml (hunk 1 above)
Comment (Vishal14583) [reply azure-pipelines.yml]: This condition was commented to verift the changes for QA env. Uncommented now.

**Review by ajasonca — APPROVED** [review]
Looks good

**Pull request merged**